File: InternalLLMBasedInspectionTool.Backend/InternalLLMBasedInspectionTool.WebApi/Models/Users/PromptSettingsModel.cs
namespace InternalLLMBasedInspectionTool.WebApi.Models.Users;

public record PromptSettingsModel {
    public required bool CheckReadability { get; init; }
    public required bool CheckPerformance { get; init; }
    public required bool CheckCorrectness { get; init; }
    public required bool CheckStyle { get; init; }
    public required bool CheckMaintainability { get; init; }
    public required bool CheckSecurity { get; init; }
    public required bool DetectLargeExpressions { get; init; }
    public required bool DetectNestedConditions { get; init; }
    public required bool EvaluateMemoryAllocations { get; init; }
    public required double MinConfidenceThreshold { get; init; }
    public required double DeterminismLevel { get; init; }
}

File: InternalLLMBasedInspectionTool.Frontend/src/app/models/users/prompt-settings.model.ts
export interface PromptSettings {
    checkReadability: boolean;
    checkPerformance: boolean;
    checkCorrectness: boolean;
    checkStyle: boolean;
    checkMaintainability: boolean;
    checkSecurity: boolean;
    detectLargeExpressions: boolean;
    detectNestedConditions: boolean;
    evaluateMemoryAllocations: boolean;
    minConfidenceThreshold: number;
    determinismLevel: number;
}
